app/presenters/flow_presenter.rb

```diff
@@ -94,6 +94,27 @@ def start_node
     @start_node ||= StartNodePresenter.new(node)
   end
 
+  def previous_state
+    unless current_state.path.empty?
+      presenter_for(@flow.node(current_state.path.last))
+    end
```

Comment:
A guard clause may be a bit cleaner here. 

```suggestion
    return if current_state.path.empty?
    
    presenter_for(@flow.node(current_state.path.last))
```

Reply:
👍